tools/coverity/misra.config

```diff
@@ -17,6 +17,18 @@
         {
             deviation: "Rule 3.1",
             reason: "Allow nested comments. Documentation blocks contain comments for example code."
+        },
+        {
+            deviation: "Rule 8.7",
+            reason: "API functions are not used by library. They must be externally visible in order to be used by the application."
+        },
+        {
+            deviation: "Rule 21.1",
```

Comment:
Where/why do we break this rule?

Reply:
Including assert.h, or defining a macro with the name assert() sets this off.
Then additionally [BackoffAlgorithm_GetNextBackoff()](https://github.com/FreeRTOS/backoffAlgorithm/blob/main/source/backoff_algorithm.c#L38) and [BackoffAlgorithm_InitializeParams](https://github.com/FreeRTOS/backoffAlgorithm/blob/main/source/backoff_algorithm.c#L84) set off rule 8.7

Reply:
instead of defining a macro with the name assert, we can use a different name and change the locations where is assert is called to instead use the new macro.